test/integration/operator_test.go

```diff
@@ -0,0 +1,231 @@
+		if errNamespace != nil {
+			t.Fatal(errNamespace)
+		}
+
+		retryErr := retry.Do(func() error {
+			_, getErr := clientset.CoreV1().Namespaces().Get(context.Background(), namespaceName, metav1.GetOptions{})
+			if !errs.IsNotFound(getErr) {
+				return errors.New("Waiting for namespace "+namespaceName+" to be deleted")
+			}
+			return nil
+		}, retry.Attempts(10), retry.MaxDelay(3*time.Second))
+
+		if retryErr != nil {
+			t.Log(retryErr)
+		}
+	})
+
+	kac := createKAC(namespaceName, resources)
+	gvr := kubearchivev1alpha1.GroupVersion.WithResource("kubearchiveconfigs")
+	_, err := dynaclient.Resource(gvr).Namespace(namespaceName).Create(context.Background(), kac, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = checkResourcesAfterApply(namespaceName, testName)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = dynaclient.Resource(gvr).Namespace(namespaceName).Delete(context.Background(), kacName, metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = checkResourcesAfterDelete(namespaceName, testName)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func checkResourcesAfterApply(namespace string, testName Test) error {
+	clientset, dynaclient, _ := test.GetKubernetesClient()
+
+	err := retry.Do(func() error {
+		gvr := schema.GroupVersionResource{Group: "sources.knative.dev", Version: "v1", Resource: "apiserversources"}
+		_, getErr := dynaclient.Resource(gvr).Namespace(test.K9eNamespace).Get(context.Background(), a13eName, metav1.GetOptions{})
+		if testName == emptyKAC {
+			if !errs.IsNotFound(getErr) {
+				return errors.New("Unexpectedly found an ApiServerSource.")
+			}
+		} else {
+			if getErr != nil {
+				return getErr
+			}
+		}
+		_, getErr = clientset.CoreV1().ConfigMaps(test.K9eNamespace).Get(context.Background(), filtersName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.CoreV1().ServiceAccounts(test.K9eNamespace).Get(context.Background(), a13eName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().ClusterRoles().Get(context.Background(), a13eName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().RoleBindings(namespace).Get(context.Background(), a13eName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().Roles(namespace).Get(context.Background(), sinkName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		_, getErr = clientset.RbacV1().RoleBindings(namespace).Get(context.Background(), sinkName, metav1.GetOptions{})
+		if getErr != nil {
+			return getErr
+		}
+		return nil
+	}, retry.Attempts(10), retry.MaxDelay(3*time.Second))
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkResourcesAfterDelete(namespace string, testName Test) error {
+	clientset, dynaclient, _ := test.GetKubernetesClient()
```

Comment:
```suggestion
func checkResourcesAfterDelete(t testing.TB, namespace string, testName Test) error {
	t.Helper()
	
	clientset, dynaclient, _ := test.GetKubernetesClient()
```

for the same reason as `checkResourcesAfterApply` because you can have `checkResourcesAfterDelete` call `t.Fatal` or `assert.<whatever> directly